sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -0,0 +1,572 @@
+        self._port = self.instance.get('port', 30015)
+        self._username = self.instance.get('username', '')
+        self._password = self.instance.get('password', '')
+        self._timeout = float(self.instance.get('timeout', 10))
+        self._batch_size = int(self.instance.get('batch_size', 1000))
+        self._tags = self.instance.get('tags', [])
+
+        # Add server & port tags
+        self._tags.append('server:{}'.format(self._server))
+        self._tags.append('port:{}'.format(self._port))
+
+        custom_queries = self.instance.get('custom_queries', [])
+        use_global_custom_queries = self.instance.get('use_global_custom_queries', True)
+
+        # Handle overrides
+        if use_global_custom_queries == 'extend':
+            custom_queries.extend(self.init_config.get('global_custom_queries', []))
+        elif 'global_custom_queries' in self.init_config and is_affirmative(use_global_custom_queries):
+            custom_queries = self.init_config.get('global_custom_queries', [])
+
+        # Deduplicate
+        self._custom_queries = list(iter_unique(custom_queries))
+
+        # We'll connect on the first check run
+        self._conn = None
+
+        # Whether or not the connection was lost
+        self._need_reconnect = False
+
+        # Whether or not to use the hostnames contained in the queried views
+        self._use_hana_hostnames = is_affirmative(self.instance.get('use_hana_hostnames', False))
+
+        # Save master database hostname to act as the default if `use_hana_hostnames` is true
+        self._master_hostname = None
+
+    def check(self, instance):
+        if self._conn is None:
+            connection = self.get_connection()
+            if connection is None:
+                return
+
+            self._conn = connection
+
+        try:
+            for query_method in (
+                self.query_master_database,
+                self.query_database_status,
+                self.query_backup_status,
+                self.query_licenses,
+                self.query_connection_overview,
+                self.query_disk_usage,
+                self.query_service_memory,
+                self.query_service_component_memory,
+                self.query_row_store_memory,
+                self.query_service_statistics,
+                self.query_volume_io,
+                self.query_custom,
+            ):
+                try:
+                    query_method()
+                except QueryExecutionError as e:
+                    self.log.error('Error querying %s: %s', e.source(), str(e))
+                    continue
+                except Exception as e:
+                    self.log.error('Unexpected error running `%s`: %s', query_method.__name__, str(e))
+                    continue
+        finally:
+            if self._need_reconnect:
+                self._conn.close()
+                self._conn = None
+                self._need_reconnect = False
```

Comment:
(nit) seems a bit confusing: "when need_reconnect is True, we close the connection and set set need_reconnect to False"

Maybe `self._connection_lost` instead of `self._need_reconnect`